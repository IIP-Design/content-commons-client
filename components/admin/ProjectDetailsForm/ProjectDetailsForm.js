/**
 *
 * ProjectDetailsForm
 *
 */
import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import {
  Button,
  Form,
  Grid,
  Input,
  TextArea,
} from 'semantic-ui-react';
import VisibilityDropdown from 'components/admin/dropdowns/VisibilityDropdown/VisibilityDropdown';
import CategoryDropdown from 'components/admin/dropdowns/CategoryDropdown/CategoryDropdown';
import CopyrightDropdown from 'components/admin/dropdowns/CopyrightDropdown/CopyrightDropdown';
import IconPopup from 'components/popups/IconPopup/IconPopup';
import UserDropdown from 'components/admin/dropdowns/UserDropdown/UserDropdown';
import TagDropdown from 'components/admin/dropdowns/TagDropdown/TagDropdown';
import TeamDropdown from 'components/admin/dropdowns/TeamDropdown/TeamDropdown';
import TermsConditions from 'components/admin/TermsConditions/TermsConditions';
import FormikAutoSave from 'components/admin/FormikAutoSave/FormikAutoSave';
import './ProjectDetailsForm.scss';

import isEmpty from 'lodash/isEmpty';

/**
 * Form component only.  Form is wrapped in specifc content type HOC to handle
 * queries, validation, etc
 * @param {object} props
 */
const ProjectDetailsForm = props => {
  const {
    values,
    errors,
    touched,
    handleSubmit,
    setFieldValue,
    isValid,
    setFieldTouched,
    maxCategories,
    status,
    save,
    config,
  } = props;

  // If form values have been updated then touched obj will have props
  // Used to prevent auto save on initial render
  const hasFormBeenUpdated = !!Object.keys( touched ).length;

  // Is this a newly created project? Used for conditionally disabling "Save draft & upload.." button - ln. 208
  const initProjectCreation = isEmpty( touched );

  const handleOnChange = ( e, {
    name, value, type, checked,
  } ) => {
    if ( type === 'checkbox' ) {
      setFieldValue( name, checked );
    } else {
      setFieldValue( name, value );
    }
    setFieldTouched( name, true, false );
  };

  const renderTeamComponent = () => {
    let TeamComponent = Form.Field;
    const baseProps = {
      id: 'team',
      control: Input,
      name: 'team',
      label: config.team?.label || 'Team',
      value: values.team,
      error: touched.team && !!errors.visibility,
      required: config.team?.required || false,
    };

    if ( config.team.dropdown ) {
      TeamComponent = TeamDropdown;
      baseProps.onChange = handleOnChange;
      baseProps.variables = config.team.variables;
      delete baseProps.control;
    }

    return <TeamComponent { ...baseProps } />;
  };

  return (
    <Fragment>
      { /* Only use autosave with existing project */ }
      { props.id && hasFormBeenUpdated && <FormikAutoSave save={ save } /> }
      <Form className="edit-project__form project-data" onSubmit={ handleSubmit }>
        <Grid stackable>
          <Grid.Row>
            <Grid.Column width="16">
              <h2 className="heading">
                <span className="uppercase">
                  { config?.headline || 'Project Data' }
                </span>
                <br />
                <small className="msg--required">Required Fields *</small>
              </h2>
            </Grid.Column>
          </Grid.Row>

          <Grid.Row>
            <Grid.Column mobile={ 16 } computer={ 8 }>

              <Form.Group widths="equal">
                { config.projectTitle
                  && (
                    <div className="field">
                      <Form.Field
                        id="projectTitle"
                        name="projectTitle"
                        control={ Input }
                        label={ config.projectTitle?.label || 'Project Title' }
                        { ...( config.projectTitle?.required && { required: true } ) }
                        autoFocus // eslint-disable-line jsx-a11y/no-autofocus
                        value={ values.projectTitle }
                        onChange={ handleOnChange }
                        error={ touched.projectTitle && !!errors.projectTitle }
                      />
                      <p className="error-message">{ touched.projectTitle ? errors.projectTitle : '' }</p>
                    </div>
                  ) }

                { config.visibility
                  && (
                    <div className="field">
                      <div className="field">
                        <VisibilityDropdown
                          id="visibility"
                          name="visibility"
                          label={ config.visibility?.label || 'Visibility Setting' }
                          value={ values.visibility }
                          onChange={ handleOnChange }
                          error={ touched.visibility && !!errors.visibility }
                          { ...( config.visibility?.required && { required: true } ) }
                          { ...( config.visibility?.props && { ...config.visibility.props } ) }
                        />
                      </div>
                      <p className="error-message">{ touched.visibility ? errors.visibility : '' }</p>
                    </div>
                  ) }
              </Form.Group>

              <Form.Group widths="equal">
                { config.author
                  && (
                    <UserDropdown
                      id="author"
                      name="author"
                      label={ config.author?.label || 'Author' }
                      value={ values.author }
                      { ...( config.author?.required && { required: true } ) }
                      onChange={ handleOnChange }
                    />
                  ) }

                { config.team && renderTeamComponent() }

                { config.copyright
                  && (
                    <div className="field">
                      <CopyrightDropdown
                        id="copyright"
                        name="copyright"
                        label={ config.copyright?.label || 'Copyright' }
                        value={ values.copyright }
                        onChange={ handleOnChange }
                        error={ touched.copyright && !!errors.copyright }
                        { ...( config.copyright?.required && { required: true } ) }
                      />
                      <p className="error-message">{ touched.copyright ? errors.copyright : '' }</p>
                    </div>
                  ) }
              </Form.Group>

              <Form.Group widths="equal">
                { config.categories
                  && (
                    <div className="field">
                      <CategoryDropdown
                        id="categories"
                        name="categories"
                        label={ config.categories?.label || 'Categories' }
                        value={ values.categories }
                        onChange={ handleOnChange }
                        error={ touched.categories && ( !values.categories.length || values.categories.length > 2 ) }
                        multiple
                        search
                        closeOnBlur
                        closeOnChange
                        { ...( config.categories?.required && { required: true } ) }
                      />

                      { errors.categories
                        ? <p className="error-message">{ touched.categories ? errors.categories : '' }</p>
                        : (
                          <p className="field__helper-text">
                            Select up to
                            { ' ' }
                            { maxCategories }
                            .
                          </p>
                        ) }
                    </div>
                  ) }

                { config.tags
                  && (
                    <div className="field">
                      <TagDropdown
                        id="tags"
                        label={ config.tags?.label || 'Tags' }
                        name="tags"
                        locale="en-us"
                        value={ values.tags }
                        onChange={ handleOnChange }
                        { ...( config.tags?.required && { required: true } ) }
                      />
                      <p className="field__helper-text">Enter keywords separated by commas to search available tags.</p>
                    </div>
                  ) }
              </Form.Group>
            </Grid.Column>

            <Grid.Column mobile={ 16 } computer={ 8 }>
              { config.descPublic
                && (
                  <Form.Field
                    id="descPublic"
                    name="descPublic"
                    control={ TextArea }
                    label={ config.descPublic?.label || 'Public Description' }
                    value={ values.descPublic }
                    onChange={ handleOnChange }
                    { ...( config.descPublic?.required && { required: true } ) }
                  />
                ) }

              { config.descInternal
                && (
                  <div className="field">
                    <Form.Field
                      className="with-helper-text"
                      id="descInternal"
                      name="descInternal"
                      control={ TextArea }
                      label={ config.descInternal?.label || 'Internal Description' }
                      value={ values.descInternal }
                      onChange={ handleOnChange }
                      { ...( config.descInternal?.required && { required: true } ) }
                    />
                    <p className="field__helper-text">
                      Share with DoS colleagues reasons for this project as it relates to Department objectives,
                      best practices for using the project, or any other notes you would like to keep internal.
                    </p>
                  </div>
                ) }

              { config.alt
                && (
                  <div className="field">
                    <Form.Field
                      className="with-helper-text"
                      id="alt"
                      name="alt"
                      control={ TextArea }
                      label={ (
                        <label htmlFor="alt">
                          { config.alt?.label || 'Alt (Alternative) Text' }
                          <IconPopup
                            message="This text helps screen-reading tools describe images to visually impaired readers."
                            iconSize="small"
                            iconType="info circle"
                            popupSize="mini"
                          />
                        </label>
                      ) }
                      value={ values.alt }
                      onChange={ handleOnChange }
                      { ...( config.alt?.required && { required: true } ) }
                    />
                  </div>
                ) }
            </Grid.Column>
          </Grid.Row>

          { !props.id && status !== 'CREATED'
          && (
            <Grid.Row reversed="computer">
              <Grid.Column mobile={ 11 }>
                <TermsConditions
                  handleOnChange={ handleOnChange }
                  error={ touched.termsConditions && !!errors.termsConditions }
                />
              </Grid.Column>
              <Grid.Column mobile={ 16 } computer={ 5 }>
                <Button
                  type="submit"
                  onClick={ handleSubmit }
                  className="edit-project__form--save"
                  content="Save draft & upload files to this project"
                  disabled={ !isValid || initProjectCreation }
                />
              </Grid.Column>
            </Grid.Row>
          ) }
        </Grid>
      </Form>
    </Fragment>
  );
};

ProjectDetailsForm.propTypes = {
  id: PropTypes.string,
  config: PropTypes.object,
  status: PropTypes.string,
  handleSubmit: PropTypes.func,
  values: PropTypes.object,
  errors: PropTypes.object,
  touched: PropTypes.object,
  setFieldValue: PropTypes.func,
  isValid: PropTypes.bool,
  setFieldTouched: PropTypes.func,
  maxCategories: PropTypes.number,
  save: PropTypes.func,
};

export default ProjectDetailsForm;
