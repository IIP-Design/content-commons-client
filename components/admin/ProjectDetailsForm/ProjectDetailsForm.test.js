import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import ProjectDetailsForm from './ProjectDetailsForm';

jest.mock( 'next/config', () => () => ( {
  publicRuntimeConfig: {
    REACT_APP_AWS_S3_AUTHORING_BUCKET: 's3-bucket-url',
  },
} ) );

jest.mock(
  'components/admin/FormikAutoSave/FormikAutoSave',
  () => function FormikAutoSave() { return ''; },
);

jest.mock(
  'components/admin/TermsConditions/TermsConditions',
  () => function TermsConditions() { return ''; },
);

jest.mock(
  'components/admin/dropdowns/TagDropdown/TagDropdown',
  () => function TagDropdown() { return ''; },
);

jest.mock(
  'components/admin/dropdowns/UserDropdown/UserDropdown',
  () => function UserDropdown() { return ''; },
);

jest.mock(
  'components/admin/dropdowns/CategoryDropdown/CategoryDropdown',
  () => function CategoryDropdown() { return ''; },
);

jest.mock(
  'components/admin/dropdowns/CopyrightDropdown/CopyrightDropdown',
  () => function CopyrightDropdown() { return ''; },
);

jest.mock(
  'components/admin/dropdowns/VisibilityDropdown/VisibilityDropdown',
  () => function VisibilityDropdown() { return ''; },
);

const props = {
  maxCategories: 2,
  values: {
    projectTitle: 'Just a test title',
    visibility: 'visibility-12345',
    author: 'author-aidkwi83',
    team: 'team-dfiiad',
    copyright: 'NO_COPYRIGHT',
    categories: ['category-123'],
    tags: ['tag-123'],
    descPublic: 'public description',
    descInternal: 'internal description',
    alt: 'some alt text',
  },
  errors: {},
  touched: {},
  isValid: true,
  dirty: false,
  handleSubmit: jest.fn(),
  handleChange: jest.fn(),
  setFieldValue: jest.fn(),
  setFieldTouched: jest.fn(),
  save: jest.fn(),
  config: {
    headline: 'Social Media Graphics Project Data',
    projectTitle: {
      label: 'Project Name',
      required: true,
    },
    visibility: {
      label: 'Visibility Setting',
      required: true,
    },
    team: {
      label: 'Source',
      required: false,
    },
    copyright: {
      label: 'Copyright',
      required: true,
    },
    categories: {
      label: 'Categories',
      required: true,
    },
    tags: {
      label: 'Tags',
      required: false,
    },
    descPublic: {
      label: 'Public Description',
      required: false,
    },
    descInternal: {
      label: 'Internal Description',
      required: false,
    },
    alt: {
      label: 'Alt (Alternative) Text',
      required: false,
    },
  },
};

describe( '<ProjectDetailsForm />, for graphic projects', () => {
  let Component;
  let wrapper;

  beforeEach( () => {
    Component = <ProjectDetailsForm { ...props } />;
    wrapper = mount( Component );
  } );

  it( 'renders without crashing', () => {
    expect( wrapper.exists() ).toEqual( true );
    expect( toJSON( wrapper ) ).toMatchSnapshot();
  } );

  it( 'has "edit-project__form project-data" class values', () => {
    const form = wrapper.find( 'Form' );

    expect( form.hasClass( 'edit-project__form project-data' ) )
      .toEqual( true );
  } );

  it( 'submit btn is disabled initially', () => {
    const submitBtn = wrapper.find( 'button.edit-project__form--save' );

    expect( submitBtn.prop( 'disabled' ) ).toEqual( true );
  } );

  it( 'submit btn is not disabled if touched or isValid', () => {
    const newProps = {
      ...props,
      touched: {
        projectTitle: true,
        termsConditions: true,
        categories: true,
        copyright: true,
      },
      dirty: true,
    };

    const newWrapper = mount( <ProjectDetailsForm { ...newProps } /> );
    const submitBtn = newWrapper.find( 'Button.edit-project__form--save' );

    expect( submitBtn.prop( 'disabled' ) ).toEqual( false );
  } );

  it( 'clicking submit btn calls handleSubmit once', () => {
    wrapper.simulate( 'submit' );
    expect( props.handleSubmit ).toHaveBeenCalledTimes( 1 );
  } );
} );

describe( '<ProjectDetailsForm />, for video projects', () => {
  let Component;
  let wrapper;

  const videoProps = {
    ...props,
    values: {
      projectTitle: 'Just a test title',
      visibility: 'visibility-12345',
      author: 'author-aidkwi83',
      team: 'team-dfiiad',
      categories: ['category-123'],
      tags: ['tag-123'],
      descPublic: 'public description',
      descInternal: 'internal description',
    },
    config: {
      headline: 'Project Data',
      projectTitle: {
        label: 'Project Title',
        required: true,
      },
      visibility: {
        label: 'Visibility Setting',
        required: true,
      },
      author: {
        label: 'Author',
        required: false,
      },
      team: {
        label: 'Team',
        required: false,
      },
      categories: {
        label: 'Categories',
        required: true,
      },
      tags: {
        label: 'Tags',
        required: false,
      },
      descPublic: {
        label: 'Public Description',
        required: false,
      },
      descInternal: {
        label: 'Internal Description',
        required: false,
      },
    },
  };

  beforeEach( () => {
    Component = <ProjectDetailsForm { ...videoProps } />;
    wrapper = mount( Component );
  } );

  it( 'renders without crashing', () => {
    expect( wrapper.exists() ).toEqual( true );
  } );

  it( 'renders the UserDropdown', () => {
    const userDropdown = wrapper.find( 'UserDropdown' );

    expect( userDropdown.exists() ).toEqual( true );
  } );

  it( 'does not render the CopyrightDropdown', () => {
    const copyrightDropdown = wrapper.find( 'CopyrightDropdown' );

    expect( copyrightDropdown.exists() ).toEqual( false );
  } );

  it( 'does not render the alt field', () => {
    const altField = wrapper.find( 'FormField#alt' );

    expect( altField.exists() ).toEqual( false );
  } );
} );

describe( '<ProjectDetailsForm />, if props.id has a value', () => {
  let Component;
  let wrapper;

  const newProps = {
    ...props,
    id: 'project-123',
  };

  beforeEach( () => {
    Component = <ProjectDetailsForm { ...newProps } />;
    wrapper = mount( Component );
  } );

  it( 'renders without crashing', () => {
    expect( wrapper.exists() ).toEqual( true );
  } );

  it( 'does not render the TermsConditions', () => {
    const termsConditions = wrapper.find( 'TermsConditions' );

    expect( termsConditions.exists() ).toEqual( false );
  } );

  it( 'does not render the save and upload', () => {
    const submitBtn = wrapper.find( 'Button.edit-project__form--save' );

    expect( submitBtn.exists() ).toEqual( false );
  } );
} );

describe( '<ProjectDetailsForm />, when there is a props.id and required fields are missing values', () => {
  let Component;
  let wrapper;

  const newProps = {
    ...props,
    id: 'project-123',
    errors: {
      projectTitle: 'A project title is required.',
      copyright: 'A copyright setting is required.',
      categories: 'At least 1 category is required.',
    },
    touched: {
      projectTitle: true,
      copyright: true,
      categories: true,
    },
    values: {
      projectTitle: '',
      copyright: '',
      categories: [],
    },
  };

  beforeEach( () => {
    Component = <ProjectDetailsForm { ...newProps } />;
    wrapper = mount( Component );
  } );

  it( 'renders without crashing', () => {
    expect( wrapper.exists() ).toEqual( true );
  } );

  it( 'renders an error message for the missing project title', () => {
    const projectTitleField = wrapper.find( 'FormField#projectTitle' );
    const errorMsg = wrapper.find( 'FormField#projectTitle + .error-message' );

    expect( projectTitleField.prop( 'error' ) ).toEqual( true );
    expect( errorMsg.text() ).toEqual( newProps.errors.projectTitle );
  } );

  it( 'renders an error message for the missing copyright value', () => {
    const copyrightDropdown = wrapper.find( 'CopyrightDropdown' );
    const errorMsg = wrapper.find( 'CopyrightDropdown + .error-message' );

    expect( copyrightDropdown.prop( 'error' ) ).toEqual( true );
    expect( errorMsg.text() ).toEqual( newProps.errors.copyright );
  } );

  it( 'renders an error message for the missing categories value(s)', () => {
    const categoriesDropdown = wrapper.find( 'CategoryDropdown' );
    const errorMsg = wrapper.find( 'CategoryDropdown + .error-message' );

    expect( categoriesDropdown.prop( 'error' ) ).toEqual( true );
    expect( errorMsg.text() ).toEqual( newProps.errors.categories );
  } );

  it( 'renders an error message for having over 2 categories values', () => {
    const excessiveCategoriesProps = {
      ...newProps,
      errors: {
        ...newProps.errors,
        categories: 'Maximum of 2 categories can be selected',
      },
      values: {
        ...newProps.values,
        categories: [
          '111', '222', '333',
        ],
      },
    };
    const newComponent = <ProjectDetailsForm { ...excessiveCategoriesProps } />;
    const newWrapper = mount( newComponent );
    const categoriesDropdown = newWrapper.find( 'CategoryDropdown' );
    const errorMsg = newWrapper.find( 'CategoryDropdown + .error-message' );

    expect( categoriesDropdown.prop( 'error' ) ).toEqual( true );
    expect( errorMsg.text() )
      .toEqual( excessiveCategoriesProps.errors.categories );
  } );
} );

describe( '<ProjectDetailsForm />, when there is no props.id and required fields are missing values', () => {
  let Component;
  let wrapper;

  const newProps = {
    ...props,
    errors: {
      termsConditions: 'You have to agree with our Terms of Use!',
    },
    touched: {
      termsConditions: true,
    },
  };

  beforeEach( () => {
    Component = <ProjectDetailsForm { ...newProps } />;
    wrapper = mount( Component );
  } );

  it( 'renders without crashing', () => {
    expect( wrapper.exists() ).toEqual( true );
  } );

  it( 'renders an error message for the TermsConditions', () => {
    const termsConditions = wrapper.find( 'TermsConditions' );

    expect( termsConditions.exists() ).toEqual( true );
    expect( termsConditions.prop( 'error' ) ).toEqual( true );
  } );
} );
