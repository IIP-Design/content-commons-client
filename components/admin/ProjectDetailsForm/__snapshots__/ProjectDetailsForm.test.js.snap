// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProjectDetailsForm />, for graphic projects renders without crashing 1`] = `
<ProjectDetailsForm
  config={
    Object {
      "alt": Object {
        "label": "Alt (Alternative) Text",
        "required": false,
      },
      "categories": Object {
        "label": "Categories",
        "required": true,
      },
      "copyright": Object {
        "label": "Copyright",
        "required": true,
      },
      "descInternal": Object {
        "label": "Internal Description",
        "required": false,
      },
      "descPublic": Object {
        "label": "Public Description",
        "required": false,
      },
      "headline": "Social Media Graphics Project Data",
      "projectTitle": Object {
        "label": "Project Name",
        "required": true,
      },
      "tags": Object {
        "label": "Tags",
        "required": false,
      },
      "team": Object {
        "label": "Source",
        "required": false,
      },
      "visibility": Object {
        "label": "Visibility Setting",
        "required": true,
      },
    }
  }
  dirty={false}
  errors={Object {}}
  handleChange={[MockFunction]}
  handleSubmit={[MockFunction]}
  isValid={true}
  maxCategories={2}
  save={[MockFunction]}
  setFieldTouched={[MockFunction]}
  setFieldValue={[MockFunction]}
  touched={Object {}}
  values={
    Object {
      "alt": "some alt text",
      "author": "author-aidkwi83",
      "categories": Array [
        "category-123",
      ],
      "copyright": "NO_COPYRIGHT",
      "descInternal": "internal description",
      "descPublic": "public description",
      "projectTitle": "Just a test title",
      "tags": Array [
        "tag-123",
      ],
      "team": "team-dfiiad",
      "visibility": "visibility-12345",
    }
  }
>
  <Form
    as="form"
    className="edit-project__form project-data"
    onSubmit={[MockFunction]}
  >
    <form
      className="ui form edit-project__form project-data"
      onSubmit={[Function]}
    >
      <Grid
        stackable={true}
      >
        <div
          className="ui stackable grid"
        >
          <GridRow>
            <div
              className="row"
            >
              <GridColumn
                width="16"
              >
                <div
                  className="sixteen wide column"
                >
                  <h2
                    className="heading"
                  >
                    <span
                      className="uppercase"
                    >
                      Social Media Graphics Project Data
                    </span>
                    <br />
                    <small
                      className="msg--required"
                    >
                      Required Fields *
                    </small>
                  </h2>
                </div>
              </GridColumn>
            </div>
          </GridRow>
          <GridRow>
            <div
              className="row"
            >
              <GridColumn
                computer={8}
                mobile={16}
              >
                <div
                  className="eight wide computer sixteen wide mobile column"
                >
                  <FormGroup
                    widths="equal"
                  >
                    <div
                      className="equal width fields"
                    >
                      <div
                        className="field"
                      >
                        <FormField
                          autoFocus={true}
                          control={[Function]}
                          id="projectTitle"
                          label="Project Name"
                          name="projectTitle"
                          onChange={[Function]}
                          required={true}
                          value="Just a test title"
                        >
                          <div
                            className="required field"
                          >
                            <label
                              htmlFor="projectTitle"
                            >
                              Project Name
                            </label>
                            <Input
                              autoFocus={true}
                              id="projectTitle"
                              name="projectTitle"
                              onChange={[Function]}
                              required={true}
                              type="text"
                              value="Just a test title"
                            >
                              <div
                                className="ui input"
                              >
                                <input
                                  autoFocus={true}
                                  id="projectTitle"
                                  name="projectTitle"
                                  onChange={[Function]}
                                  required={true}
                                  type="text"
                                  value="Just a test title"
                                />
                              </div>
                            </Input>
                          </div>
                        </FormField>
                        <p
                          className="error-message"
                        />
                      </div>
                      <div
                        className="field"
                      >
                        <div
                          className="field"
                        >
                          <VisibilityDropdown
                            hide="internal"
                            id="visibility"
                            label="Visibility Setting"
                            name="visibility"
                            onChange={[Function]}
                            required={true}
                            value="visibility-12345"
                          />
                        </div>
                        <p
                          className="error-message"
                        />
                      </div>
                    </div>
                  </FormGroup>
                  <FormGroup
                    widths="equal"
                  >
                    <div
                      className="equal width fields"
                    >
                      <FormField
                        control={[Function]}
                        id="team"
                        label="Source"
                        name="team"
                        placeholder=""
                        value="team-dfiiad"
                      >
                        <div
                          className="field"
                        >
                          <label
                            htmlFor="team"
                          >
                            Source
                          </label>
                          <Input
                            id="team"
                            name="team"
                            placeholder=""
                            type="text"
                            value="team-dfiiad"
                          >
                            <div
                              className="ui input"
                            >
                              <input
                                id="team"
                                name="team"
                                onChange={[Function]}
                                placeholder=""
                                type="text"
                                value="team-dfiiad"
                              />
                            </div>
                          </Input>
                        </div>
                      </FormField>
                      <div
                        className="field"
                      >
                        <CopyrightDropdown
                          id="copyright"
                          label="Copyright"
                          name="copyright"
                          onChange={[Function]}
                          required={true}
                          value="NO_COPYRIGHT"
                        />
                        <p
                          className="error-message"
                        />
                      </div>
                    </div>
                  </FormGroup>
                  <FormGroup
                    widths="equal"
                  >
                    <div
                      className="equal width fields"
                    >
                      <div
                        className="field"
                      >
                        <CategoryDropdown
                          closeOnBlur={true}
                          closeOnChange={true}
                          id="categories"
                          label="Categories"
                          multiple={true}
                          name="categories"
                          onChange={[Function]}
                          required={true}
                          search={true}
                          value={
                            Array [
                              "category-123",
                            ]
                          }
                        />
                        <p
                          className="field__helper-text"
                        >
                          Select up to 
                          2
                          .
                        </p>
                      </div>
                      <div
                        className="field"
                      >
                        <TagDropdown
                          id="tags"
                          label="Tags"
                          locale="en-us"
                          name="tags"
                          onChange={[Function]}
                          value={
                            Array [
                              "tag-123",
                            ]
                          }
                        />
                        <p
                          className="field__helper-text"
                        >
                          Enter keywords separated by commas to search available tags.
                        </p>
                      </div>
                    </div>
                  </FormGroup>
                </div>
              </GridColumn>
              <GridColumn
                computer={8}
                mobile={16}
              >
                <div
                  className="eight wide computer sixteen wide mobile column"
                >
                  <FormField
                    control={[Function]}
                    id="descPublic"
                    label="Public Description"
                    name="descPublic"
                    onChange={[Function]}
                    value="public description"
                  >
                    <div
                      className="field"
                    >
                      <label
                        htmlFor="descPublic"
                      >
                        Public Description
                      </label>
                      <TextArea
                        as="textarea"
                        id="descPublic"
                        name="descPublic"
                        onChange={[Function]}
                        rows={3}
                        value="public description"
                      >
                        <Ref
                          innerRef={
                            Object {
                              "current": <textarea
                                id="descPublic"
                                name="descPublic"
                                rows="3"
                              >
                                public description
                              </textarea>,
                            }
                          }
                        >
                          <RefFindNode
                            innerRef={
                              Object {
                                "current": <textarea
                                  id="descPublic"
                                  name="descPublic"
                                  rows="3"
                                >
                                  public description
                                </textarea>,
                              }
                            }
                          >
                            <textarea
                              id="descPublic"
                              name="descPublic"
                              onChange={[Function]}
                              onInput={[Function]}
                              rows={3}
                              value="public description"
                            />
                          </RefFindNode>
                        </Ref>
                      </TextArea>
                    </div>
                  </FormField>
                  <div
                    className="field"
                  >
                    <FormField
                      className="with-helper-text"
                      control={[Function]}
                      id="descInternal"
                      label="Internal Description"
                      name="descInternal"
                      onChange={[Function]}
                      value="internal description"
                    >
                      <div
                        className="field with-helper-text"
                      >
                        <label
                          htmlFor="descInternal"
                        >
                          Internal Description
                        </label>
                        <TextArea
                          as="textarea"
                          id="descInternal"
                          name="descInternal"
                          onChange={[Function]}
                          rows={3}
                          value="internal description"
                        >
                          <Ref
                            innerRef={
                              Object {
                                "current": <textarea
                                  id="descInternal"
                                  name="descInternal"
                                  rows="3"
                                >
                                  internal description
                                </textarea>,
                              }
                            }
                          >
                            <RefFindNode
                              innerRef={
                                Object {
                                  "current": <textarea
                                    id="descInternal"
                                    name="descInternal"
                                    rows="3"
                                  >
                                    internal description
                                  </textarea>,
                                }
                              }
                            >
                              <textarea
                                id="descInternal"
                                name="descInternal"
                                onChange={[Function]}
                                onInput={[Function]}
                                rows={3}
                                value="internal description"
                              />
                            </RefFindNode>
                          </Ref>
                        </TextArea>
                      </div>
                    </FormField>
                    <p
                      className="field__helper-text"
                    >
                      Share with DoS colleagues reasons for this project as it relates to Department objectives, best practices for using the project, or any other notes you would like to keep internal.
                    </p>
                  </div>
                  <div
                    className="field"
                  >
                    <FormField
                      className="with-helper-text"
                      control={[Function]}
                      id="alt"
                      label={
                        <label
                          htmlFor="alt"
                        >
                          Alt (Alternative) Text
                          <IconPopup
                            iconSize="small"
                            iconType="info circle"
                            message="This text helps screen-reading tools describe images to visually impaired readers."
                            popupSize="mini"
                          />
                        </label>
                      }
                      name="alt"
                      onChange={[Function]}
                      value="some alt text"
                    >
                      <div
                        className="field with-helper-text"
                      >
                        <label
                          htmlFor="alt"
                        >
                          Alt (Alternative) Text
                          <IconPopup
                            iconSize="small"
                            iconType="info circle"
                            message="This text helps screen-reading tools describe images to visually impaired readers."
                            popupSize="mini"
                          >
                            <Popup
                              content="This text helps screen-reading tools describe images to visually impaired readers."
                              disabled={false}
                              eventsEnabled={true}
                              inverted={true}
                              offset={0}
                              on={
                                Array [
                                  "hover",
                                  "click",
                                  "focus",
                                ]
                              }
                              pinned={false}
                              position="top left"
                              size="mini"
                              trigger={
                                <span>
                                  <Focusable
                                    order={0}
                                  >
                                    <Icon
                                      as="i"
                                      name="info circle"
                                      size="small"
                                    />
                                  </Focusable>
                                </span>
                              }
                            >
                              <Portal
                                closeOnDocumentClick={true}
                                closeOnEscape={true}
                                closeOnTriggerBlur={true}
                                closeOnTriggerClick={true}
                                closeOnTriggerMouseLeave={true}
                                eventPool="default"
                                mouseEnterDelay={50}
                                mouseLeaveDelay={70}
                                onClose={[Function]}
                                onMount={[Function]}
                                onOpen={[Function]}
                                onUnmount={[Function]}
                                openOnTriggerClick={true}
                                openOnTriggerFocus={true}
                                openOnTriggerMouseEnter={true}
                                trigger={
                                  <span>
                                    <Focusable
                                      order={0}
                                    >
                                      <Icon
                                        as="i"
                                        name="info circle"
                                        size="small"
                                      />
                                    </Focusable>
                                  </span>
                                }
                                triggerRef={
                                  Object {
                                    "current": <span>
                                      <span
                                        class="focusable"
                                        tabindex="0"
                                      >
                                        <i
                                          aria-hidden="true"
                                          class="info circle small icon"
                                        />
                                      </span>
                                    </span>,
                                  }
                                }
                              >
                                <Ref
                                  innerRef={[Function]}
                                >
                                  <RefFindNode
                                    innerRef={[Function]}
                                  >
                                    <span
                                      onBlur={[Function]}
                                      onClick={[Function]}
                                      onFocus={[Function]}
                                      onMouseEnter={[Function]}
                                      onMouseLeave={[Function]}
                                    >
                                      <Focusable
                                        order={0}
                                      >
                                        <span
                                          className="focusable"
                                          tabIndex={0}
                                        >
                                          <Icon
                                            as="i"
                                            name="info circle"
                                            size="small"
                                          >
                                            <i
                                              aria-hidden="true"
                                              className="info circle small icon"
                                              onClick={[Function]}
                                            />
                                          </Icon>
                                        </span>
                                      </Focusable>
                                    </span>
                                  </RefFindNode>
                                </Ref>
                              </Portal>
                            </Popup>
                          </IconPopup>
                        </label>
                        <TextArea
                          as="textarea"
                          id="alt"
                          name="alt"
                          onChange={[Function]}
                          rows={3}
                          value="some alt text"
                        >
                          <Ref
                            innerRef={
                              Object {
                                "current": <textarea
                                  id="alt"
                                  name="alt"
                                  rows="3"
                                >
                                  some alt text
                                </textarea>,
                              }
                            }
                          >
                            <RefFindNode
                              innerRef={
                                Object {
                                  "current": <textarea
                                    id="alt"
                                    name="alt"
                                    rows="3"
                                  >
                                    some alt text
                                  </textarea>,
                                }
                              }
                            >
                              <textarea
                                id="alt"
                                name="alt"
                                onChange={[Function]}
                                onInput={[Function]}
                                rows={3}
                                value="some alt text"
                              />
                            </RefFindNode>
                          </Ref>
                        </TextArea>
                      </div>
                    </FormField>
                  </div>
                </div>
              </GridColumn>
            </div>
          </GridRow>
          <GridRow
            reversed="computer"
          >
            <div
              className="computer reversed row"
            >
              <GridColumn
                mobile={11}
              >
                <div
                  className="eleven wide mobile column"
                >
                  <TermsConditions
                    handleOnChange={[Function]}
                  />
                </div>
              </GridColumn>
              <GridColumn
                computer={5}
                mobile={16}
              >
                <div
                  className="five wide computer sixteen wide mobile column"
                >
                  <Button
                    as="button"
                    className="edit-project__form--save"
                    content="Save draft & upload files to this project"
                    disabled={true}
                    onClick={[MockFunction]}
                    type="submit"
                  >
                    <Ref
                      innerRef={
                        Object {
                          "current": <button
                            class="ui disabled button edit-project__form--save"
                            disabled=""
                            tabindex="-1"
                            type="submit"
                          >
                            Save draft & upload files to this project
                          </button>,
                        }
                      }
                    >
                      <RefFindNode
                        innerRef={
                          Object {
                            "current": <button
                              class="ui disabled button edit-project__form--save"
                              disabled=""
                              tabindex="-1"
                              type="submit"
                            >
                              Save draft & upload files to this project
                            </button>,
                          }
                        }
                      >
                        <button
                          className="ui disabled button edit-project__form--save"
                          disabled={true}
                          onClick={[Function]}
                          tabIndex={-1}
                          type="submit"
                        >
                          Save draft & upload files to this project
                        </button>
                      </RefFindNode>
                    </Ref>
                  </Button>
                </div>
              </GridColumn>
            </div>
          </GridRow>
        </div>
      </Grid>
    </form>
  </Form>
</ProjectDetailsForm>
`;
